mono/mini/debugger-agent.c

```diff
@@ -5167,8 +5203,10 @@ ss_start (SingleStepReq *ss_req, MonoMethod *method, SeqPoint* sp, MonoSeqPointI
 			for (i = 0; i < sp->next_len; i++) {
 				next_sp = &next[i];
 
-				bp = set_breakpoint (method, next_sp->il_offset, ss_req->req, NULL);
-				ss_req->bps = g_slist_append (ss_req->bps, bp);
+				if (!ss_bp_is_duplicate(ss_req->bps, method, next_sp->il_offset)) {
```

Comment:
@vargaz, I'm going to go ahead and implement the hash table idea, however this seems to me to add dangerous complexity for an edge case. I would like to add a test case for this situation. Could you give me advice on how to reproduce this? Make a large switch and then break on the switch() itself, I guess?